<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get Started â€¢ deseq2pip</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">deseq2pip</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/deseq2pip.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/rna_guide.html">Advance Guide</a></li>
    <li><a class="dropdown-item" href="../articles/gsea_collection.html">Build GSEA Gene Set Collections</a></li>
    <li><a class="dropdown-item" href="../articles/session_info.html">Session Info</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/hungms/deseq2pip/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get Started</h1>
                        <h4 data-toc-skip class="author">Author: Matthew
Hung</h4>
            
            <h4 data-toc-skip class="date">Last Updated: 2025/04/04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hungms/deseq2pip/blob/main/vignettes/deseq2pip.Rmd" class="external-link"><code>vignettes/deseq2pip.Rmd</code></a></small>
      <div class="d-none name"><code>deseq2pip.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load deseq2pip  </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hungms.github.io/deseq2pip/">deseq2pip</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'magrittr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:rlang':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     set_names</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:tidyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     extract</span></span></code></pre>
<pre><code><span><span class="co">## strpip 0.1.0</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'cowplot'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:patchwork':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     align_plots</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: viridisLite</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine, intersect, setdiff, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:tidyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     first, rename</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'IRanges'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     collapse, desc, slice</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'GenomicRanges'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:magrittr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     subtract</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'matrixStats'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     count</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:rlang':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     exprs</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">## 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span></code></pre>
<pre><code><span><span class="co">## Registered S3 methods overwritten by 'ggalt':</span></span>
<span><span class="co">##   method                  from   </span></span>
<span><span class="co">##   grid.draw.absoluteGrob  ggplot2</span></span>
<span><span class="co">##   grobHeight.absoluteGrob ggplot2</span></span>
<span><span class="co">##   grobWidth.absoluteGrob  ggplot2</span></span>
<span><span class="co">##   grobX.absoluteGrob      ggplot2</span></span>
<span><span class="co">##   grobY.absoluteGrob      ggplot2</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'ggalt'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:rlang':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     bytes</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## clusterProfiler v4.14.6 Learn more at https://yulab-smu.top/contribution-knowledge-mining/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Please cite:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,</span></span>
<span><span class="co">## X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal</span></span>
<span><span class="co">## enrichment tool for interpreting omics data. The Innovation. 2021,</span></span>
<span><span class="co">## 2(3):100141</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'clusterProfiler'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     slice</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:S4Vectors':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rename</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:biomaRt':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     select</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter</span></span></code></pre>
<pre><code><span><span class="co">## For full functionality, please install the 'msigdbdf' package with:</span></span>
<span><span class="co">## install.packages('msigdbdf', repos = 'https://igordot.r-universe.dev')</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'ggpubr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:plotr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     get_palette</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:cowplot':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     get_legend</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'scales'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:viridis':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     viridis_pal</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'limma'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:DESeq2':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: mgcv</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: nlme</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'nlme'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     collapse</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     collapse</span></span></code></pre>
<pre><code><span><span class="co">## This is mgcv 1.9-1. For overview type 'help("mgcv-package")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: genefilter</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'genefilter'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowSds, rowVars</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowSds, rowVars</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:patchwork':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     area</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocParallel</span></span></code></pre>
<pre><code><span><span class="co">## enrichplot v1.26.6 Learn more at https://yulab-smu.top/contribution-knowledge-mining/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Please cite:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using</span></span>
<span><span class="co">## GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.</span></span>
<span><span class="co">## Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'enrichplot'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:ggpubr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     color_palette</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'deseq2pip'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:cowplot':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     save_plot</span></span></code></pre>
<div class="section level2">
<h2 id="run-deseq2pip-on-rna-seq-data">Run deseq2pip on RNA-seq data<a class="anchor" aria-label="anchor" href="#run-deseq2pip-on-rna-seq-data"></a>
</h2>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create a directory for results, currently set to tempdir()</span></span>
<span><span class="co"># save_dir &lt;- "/Users/hungm/Documents/development/deseq2pip/tests/rnaseq/pipeline/"</span></span>
<span><span class="va">save_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">save_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in dir.create(save_dir, recursive = TRUE): '/tmp/RtmpTUxHJn' already</span></span>
<span><span class="co">## exists</span></span></code></pre>
<div class="section level3">
<h3 id="load-deseq2-object-from-nfcore-pipeline">Load DESeq2 object from nfcore pipeline<a class="anchor" aria-label="anchor" href="#load-deseq2-object-from-nfcore-pipeline"></a>
</h3>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example DESeq2 object</span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE189410.dds.RData"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span></span>
<span><span class="va">tx2gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">gzfile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE189410.tx2gene.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_nfcore_rna.html">import_nfcore_rna</a></span><span class="op">(</span>rdata <span class="op">=</span> <span class="va">rdata</span>, tx2gene <span class="op">=</span> <span class="va">tx2gene</span><span class="op">)</span></span>
<span><span class="fu">assay</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##               BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1</span></span>
<span><span class="co">## 0610005C13Rik       18        8       18       11       12</span></span>
<span><span class="co">## 0610006L08Rik        0        0       11        0        0</span></span>
<span><span class="co">## 0610009B22Rik      102       93      124      128      126</span></span>
<span><span class="co">## 0610009E02Rik       50       11       22       13       21</span></span>
<span><span class="co">## 0610009L18Rik        5       10        3        4        3</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="set-group-variable-to-compare-between">Set group variable to compare between<a class="anchor" aria-label="anchor" href="#set-group-variable-to-compare-between"></a>
</h3>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set group variable to compare between, with the first level set as the reference</span></span>
<span><span class="va">dds</span><span class="op">$</span><span class="va">Group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">Group2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'IgM'</span>, <span class="st">'IgG'</span>, <span class="st">'IgA'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set dds design with desired variables</span></span>
<span><span class="fu">design</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Group2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-deseq2-pipeline">Run deseq2 pipeline<a class="anchor" aria-label="anchor" href="#run-deseq2-pipeline"></a>
</h3>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run deseq2pip</span></span>
<span><span class="fu"><a href="../reference/run_deseq2_pip.html">run_deseq2_pip</a></span><span class="op">(</span></span>
<span>    <span class="va">dds</span>,</span>
<span>    assaytype <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>    batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    experiment <span class="op">=</span> <span class="st">"GSE189410"</span>,</span>
<span>    remove_xy <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    remove_mt <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    org <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"Group2"</span>,</span>
<span>    save_dir <span class="op">=</span> <span class="va">save_dir</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running quality control pipeline...</span></span></code></pre>
<pre><code><span><span class="co">## Removing XY genes...</span></span></code></pre>
<pre><code><span><span class="co">## Removing MT genes...</span></span></code></pre>
<pre><code><span><span class="co">## Filtering genes with low expressions...</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-1.png" width="700"><img src="deseq2pip_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<pre><code><span><span class="co">## Saving DESeq2 object &amp; expressions...</span></span></code></pre>
<pre><code><span><span class="co">## Running distance analysis pipeline...</span></span></code></pre>
<pre><code><span><span class="co">## Running batch correction...</span></span></code></pre>
<pre><code><span><span class="co">## Running PCA and distance analysis...</span></span></code></pre>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-3.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-4.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-5.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-6.png" width="700"><img src="deseq2pip_files/figure-html/unnamed-chunk-5-7.png" width="700"></p>
<pre><code><span><span class="co">## Running differential expression analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Running DESeq2 differential expression...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating MA plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 1096 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 1694 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 768 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating volcano plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 36490 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 53097 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 37 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 36490 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 53097 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 35 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-8.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 35583 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 52390 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 30 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 35583 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 52390 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 23 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-9.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 38113 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 53040 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 39 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 38113 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 53040 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 36 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-10.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Running GSEA pipeline...</span></span></code></pre>
<pre><code><span><span class="co">## Running gene set enrichment analysis...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.18% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.33% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.6% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating GSEA barplots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-11.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |====                                                                  |   6%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-12.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |========                                                              |  11%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-13.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |============                                                          |  17%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-14.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |================                                                      |  22%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-15.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================                                                   |  28%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-16.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-17.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===========================                                           |  39%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-18.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================                                       |  44%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-19.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================================                                   |  50%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-20.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================================                               |  56%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-21.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===========================================                           |  61%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-22.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-23.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================================================                   |  72%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-24.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================                |  78%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-25.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==========================================================            |  83%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-26.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==============================================================        |  89%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-27.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==================================================================    |  94%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-5-28.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Formatting for enrichmentmap...</span></span></code></pre>
<pre><code><span><span class="co">## Complete; returning DESeq2 object...</span></span></code></pre>
<pre><code><span><span class="co">## class: DESeqDataSet </span></span>
<span><span class="co">## dim: 53571 11 </span></span>
<span><span class="co">## metadata(1): version</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(53571): 0610005C13Rik 0610006L08Rik ... snoZ159 snoZ196</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(11): BM_IgA_1 BM_IgA_2 ... BM_IgM_3 BM_IgM_4</span></span>
<span><span class="co">## colData names(6): sample Group1 ... sizeFactor condition</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="run-deseq2pip-on-atac-seq-data">Run deseq2pip on ATAC-seq data<a class="anchor" aria-label="anchor" href="#run-deseq2pip-on-atac-seq-data"></a>
</h2>
<div class="sourceCode" id="cb319"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create a directory for results, currently set to tempdir()</span></span>
<span><span class="co"># save_dir &lt;- "/Users/hungm/Documents/development/deseq2pip/tests/atacseq/pipeline/"</span></span>
<span><span class="va">save_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">save_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in dir.create(save_dir, recursive = TRUE): '/tmp/RtmpTUxHJn' already</span></span>
<span><span class="co">## exists</span></span></code></pre>
<div class="section level3">
<h3 id="load-deseq2-object-from-nfcore-pipeline-1">Load DESeq2 object from nfcore pipeline<a class="anchor" aria-label="anchor" href="#load-deseq2-object-from-nfcore-pipeline-1"></a>
</h3>
<div class="sourceCode" id="cb321"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example DESeq2 object</span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE224512.dds.RData"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span></span>
<span><span class="va">annotatePeaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">gzfile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE224512.annotatePeaks.txt.gz"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_nfcore_atac.html">import_nfcore_atac</a></span><span class="op">(</span>rdata <span class="op">=</span> <span class="va">rdata</span>, annotatePeaks <span class="op">=</span> <span class="va">annotatePeaks</span><span class="op">)</span></span>
<span><span class="fu">assay</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            CC_REP3 CB_REP1 CC_REP4 CC_REP1 CC_REP2</span></span>
<span><span class="co">## Interval_1      13       9      12      14      18</span></span>
<span><span class="co">## Interval_2      16      13      11      15      10</span></span>
<span><span class="co">## Interval_3      16       9      14       9      15</span></span>
<span><span class="co">## Interval_4       7      16      17      20      16</span></span>
<span><span class="co">## Interval_5       3       4       4       6       6</span></span></code></pre>
<div class="sourceCode" id="cb323"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># subset groups for brevity</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span>, <span class="va">dds</span><span class="op">$</span><span class="va">Group1</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">'BC'</span>, <span class="st">'BCK'</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="set-group-variable-to-compare-between-1">Set group variable to compare between<a class="anchor" aria-label="anchor" href="#set-group-variable-to-compare-between-1"></a>
</h3>
<div class="sourceCode" id="cb324"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set group variable to compare between, with the first level set as the reference</span></span>
<span><span class="va">dds</span><span class="op">$</span><span class="va">Group1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">Group1</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">'BC'</span>, <span class="st">'BCK'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set dds design with desired variables</span></span>
<span><span class="fu">design</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Group1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-deseq2-pipeline-1">Run deseq2 pipeline<a class="anchor" aria-label="anchor" href="#run-deseq2-pipeline-1"></a>
</h3>
<div class="sourceCode" id="cb325"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run deseq2pip</span></span>
<span><span class="fu"><a href="../reference/run_deseq2_pip.html">run_deseq2_pip</a></span><span class="op">(</span></span>
<span>    <span class="va">dds</span>,</span>
<span>    assaytype <span class="op">=</span> <span class="st">"ATAC"</span>,</span>
<span>    batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    experiment <span class="op">=</span> <span class="st">"GSE224512"</span>,</span>
<span>    remove_xy <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    remove_mt <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    org <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"Group1"</span>,</span>
<span>    save_dir <span class="op">=</span> <span class="va">save_dir</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running quality control pipeline...</span></span></code></pre>
<pre><code><span><span class="co">## Removing XY genes...</span></span></code></pre>
<pre><code><span><span class="co">## Removing MT genes...</span></span></code></pre>
<pre><code><span><span class="co">## Filtering genes with low expressions...</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-1.png" width="700"><img src="deseq2pip_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<pre><code><span><span class="co">## Saving DESeq2 object &amp; expressions...</span></span></code></pre>
<pre><code><span><span class="co">## Running distance analysis pipeline...</span></span></code></pre>
<pre><code><span><span class="co">## Running batch correction...</span></span></code></pre>
<pre><code><span><span class="co">## Running PCA and distance analysis...</span></span></code></pre>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-3.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-4.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-5.png" width="700"><img src="deseq2pip_files/figure-html/unnamed-chunk-9-6.png" width="700"></p>
<pre><code><span><span class="co">## Running differential expression analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Running DESeq2 differential expression...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating MA plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 340 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 3651 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 6951 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating volcano plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2824 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72613 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2824 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72613 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-7.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 23960 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72498 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 32 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 23960 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72498 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 18 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-8.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 9870 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72626 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 32 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 9870 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 72626 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 30 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-9.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Repeating analysis for TSS peaks...</span></span></code></pre>
<pre><code><span><span class="co">## Running batch correction...</span></span></code></pre>
<pre><code><span><span class="co">## Running PCA and distance analysis...</span></span></code></pre>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-10.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-11.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-12.png" width="700"></p>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-13.png" width="700"><img src="deseq2pip_files/figure-html/unnamed-chunk-9-14.png" width="700"></p>
<pre><code><span><span class="co">## Running DESeq2 differential expression...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating MA plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 14 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 1600 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 1875 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating volcano plots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 248 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12717 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 248 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12717 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-15.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1234 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12677 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 7 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1234 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12677 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 6 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-16.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 247 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12717 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 247 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 12717 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-17.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Running gene set enrichment analysis...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : There were 3 pathways for which P-values were not calculated</span></span>
<span><span class="co">## properly due to unbalanced (positive and negative) gene-level statistic values.</span></span>
<span><span class="co">## For such pathways pval, padj, NES, log2err are set to NA. You can try to</span></span>
<span><span class="co">## increase the value of the argument nPermSimple (for example set it nPermSimple</span></span>
<span><span class="co">## = 10000)</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some of the pathways the P-values were likely overestimated. For</span></span>
<span><span class="co">## such pathways log2err is set to NA.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some of the pathways the P-values were likely overestimated. For</span></span>
<span><span class="co">## such pathways log2err is set to NA.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span><span class="co">## set the `eps` argument to zero for better estimation.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : There were 5 pathways for which P-values were not calculated</span></span>
<span><span class="co">## properly due to unbalanced (positive and negative) gene-level statistic values.</span></span>
<span><span class="co">## For such pathways pval, padj, NES, log2err are set to NA. You can try to</span></span>
<span><span class="co">## increase the value of the argument nPermSimple (for example set it nPermSimple</span></span>
<span><span class="co">## = 10000)</span></span>
<span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span><span class="co">## set the `eps` argument to zero for better estimation.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : There were 2 pathways for which P-values were not calculated</span></span>
<span><span class="co">## properly due to unbalanced (positive and negative) gene-level statistic values.</span></span>
<span><span class="co">## For such pathways pval, padj, NES, log2err are set to NA. You can try to</span></span>
<span><span class="co">## increase the value of the argument nPermSimple (for example set it nPermSimple</span></span>
<span><span class="co">## = 10000)</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : There were 2 pathways for which P-values were not calculated</span></span>
<span><span class="co">## properly due to unbalanced (positive and negative) gene-level statistic values.</span></span>
<span><span class="co">## For such pathways pval, padj, NES, log2err are set to NA. You can try to</span></span>
<span><span class="co">## increase the value of the argument nPermSimple (for example set it nPermSimple</span></span>
<span><span class="co">## = 10000)</span></span></code></pre>
<pre><code><span><span class="co">## Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span><span class="co">## minSize, : For some of the pathways the P-values were likely overestimated. For</span></span>
<span><span class="co">## such pathways log2err is set to NA.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Generating GSEA barplots...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-18.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |====                                                                  |   6%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-19.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |========                                                              |  11%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-20.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |============                                                          |  17%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-21.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |================                                                      |  22%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-22.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================                                                   |  28%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-23.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-24.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===========================                                           |  39%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-25.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================                                       |  44%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-26.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================================                                   |  50%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-27.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |=======================================                               |  56%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-28.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===========================================                           |  61%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-29.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-30.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |===================================================                   |  72%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"): Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-31.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================                |  78%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-32.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==========================================================            |  83%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-33.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==============================================================        |  89%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 7 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 7 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-34.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |==================================================================    |  94%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span>
<span><span class="co">## Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<p><img src="deseq2pip_files/figure-html/unnamed-chunk-9-35.png" width="700"></p>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Formatting for enrichmentmap...</span></span></code></pre>
<pre><code><span><span class="co">## TSS peak analysis complete; returning DESeq2 object...</span></span></code></pre>
<pre><code><span><span class="co">## class: DESeqDataSet </span></span>
<span><span class="co">## dim: 72693 13 </span></span>
<span><span class="co">## metadata(1): version</span></span>
<span><span class="co">## assays(2): counts vst</span></span>
<span><span class="co">## rownames(72693): Interval_1 Interval_2 ... Interval_74850</span></span>
<span><span class="co">##   Interval_74851</span></span>
<span><span class="co">## rowData names(19): Chr Start ... gene TSS</span></span>
<span><span class="co">## colnames(13): WT_REP4 BCK_REP5 ... BC_REP2 BC_REP1</span></span>
<span><span class="co">## colData names(4): sample Group1 Group2 sizeFactor</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Miu Shing Hung.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
