<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Advance Guide • deseq2pip</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Advance Guide">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">deseq2pip</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/deseq2pip.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/rna_guide.html">Advance Guide</a></li>
    <li><a class="dropdown-item" href="../articles/gsea_collection.html">Build GSEA Gene Set Collections</a></li>
    <li><a class="dropdown-item" href="../articles/session_info.html">Session Info</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/hungms/deseq2pip/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Advance Guide</h1>
                        <h4 data-toc-skip class="author">Author: Matthew
Hung</h4>
            
            <h4 data-toc-skip class="date">Last Updated: 2025/04/08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hungms/deseq2pip/blob/main/vignettes/rna_guide.Rmd" class="external-link"><code>vignettes/rna_guide.Rmd</code></a></small>
      <div class="d-none name"><code>rna_guide.Rmd</code></div>
    </div>

    
    
<p>First, load the deseq2pip package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hungms.github.io/deseq2pip/">deseq2pip</a></span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">## 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span></code></pre>
<div class="section level2">
<h2 id="data-import-and-setup">Data Import and Setup<a class="anchor" aria-label="anchor" href="#data-import-and-setup"></a>
</h2>
<p>Let’s import and prepare your data. We’ll use the example dataset
GSE189410 to demonstrate the workflow:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example DESeq2 object</span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE189410.dds.RData"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span></span>
<span><span class="va">tx2gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">gzfile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE189410.tx2gene.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"deseq2pip"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_nfcore_rna.html">import_nfcore_rna</a></span><span class="op">(</span>rdata <span class="op">=</span> <span class="va">rdata</span>, tx2gene <span class="op">=</span> <span class="va">tx2gene</span><span class="op">)</span></span>
<span><span class="fu">assay</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##               BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1</span></span>
<span><span class="co">## 0610005C13Rik       18        8       18       11       12</span></span>
<span><span class="co">## 0610006L08Rik        0        0       11        0        0</span></span>
<span><span class="co">## 0610009B22Rik      102       93      124      128      126</span></span>
<span><span class="co">## 0610009E02Rik       50       11       22       13       21</span></span>
<span><span class="co">## 0610009L18Rik        5       10        3        4        3</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set group variable to compare between, with the first level set as the reference</span></span>
<span><span class="va">dds</span><span class="op">$</span><span class="va">Group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">Group2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'IgM'</span>, <span class="st">'IgG'</span>, <span class="st">'IgA'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set dds design with desired variables</span></span>
<span><span class="co"># for more information about setting contrasts in DESeq2, see: https://github.com/tavareshugo/tutorial_DESeq2_contrasts?tab=readme-ov-file</span></span>
<span><span class="fu">design</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Group2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-deseq2-pipeline">Run DESeq2 pipeline<a class="anchor" aria-label="anchor" href="#run-deseq2-pipeline"></a>
</h2>
<p>The run_deseq2_pip() function is a wrapper function that runs the
complete RNA-seq analysis for <strong>every comparison pair</strong> in
the specified group variable.</p>
<p>The analysis for each comparison pair can be broken down into three
modules:</p>
<ol style="list-style-type: decimal">
<li>Quality Control: Filters genes, generates QC/PCA/distance plots, and
saves expression data</li>
<li>Differential Expression Analysis: Performs DESeq2 analysis,
functional gene annotation, and creates volcano plots</li>
<li>GSEA Analysis: Find enriched pathways using MSigDB collections and
prepare results for EnrichmentMap visualization</li>
</ol>
<p>For our example dataset, we hqf3 3 comparison pairs in the group
variable <code>Group2</code>: <code>IgM</code> vs <code>IgG</code>,
<code>IgM</code> vs <code>IgA</code>, and <code>IgG</code> vs
<code>IgA</code>.</p>
<p>Here’s how to run the complete pipeline with a single command:
<em>Note: Repeating run_deseq2_pip will overwrite the existing files if
the same directory name is used. To prevent this, please specify a
different save_dir directory.</em></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run deseq2 pipeline</span></span>
<span><span class="co">#run_deseq2_pip(dds, experiment = "GSE189410", org = "mouse", group_by = "Group2", remove_xy = TRUE, remove_mt = TRUE, save_dir = "/Users/hungm/Documents/development/deseq2pip/tests/pipeline/")</span></span></code></pre></div>
<p>This will create a directory structure under the specified save
directory: * qc_results/ contains QC plots and expression data *
batch_corrected/ contains batch corrected plots and expression data *
<code>Group2</code>/ contains directories for each comparison pair</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show directory structure</span></span>
<span><span class="co">#fs::dir_tree("/Users/hungm/Documents/development/deseq2pip/tests/pipeline/", recurse = 1)</span></span></code></pre></div>
<p>For each comparison pair, the directory contains: *
diffexp_deseq2_wald.tsv: Dataframe of differential expression results
and gene annotation * diffexp_volcano.pdf: Volcano plot of the
differential expression results * outputs for each gene set collection
(HALLMARK, GOBP, KEGG, REACTOME, TFT, BIOCARTA): * gsea_<em>.rds: RDS
file of gseaResult class object </em> gsea_<em>.tsv: TSV file of GSEA
results </em> gsea_*_barplot.pdf: Barplots for the top N terms from GSEA
results</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show directory structure for a specific comparison pair</span></span>
<span><span class="co">#fs::dir_tree("/Users/hungm/Documents/development/deseq2pip/tests/pipeline/Group2/IgA_vs_IgG/")</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-individual-modules">Run individual modules<a class="anchor" aria-label="anchor" href="#run-individual-modules"></a>
</h2>
<p>After an initial run of the DESeq2 pipeline, you may wish to run a
specific module using different parameters or comparison groups without
repeating the entire pipeline.</p>
<p>Here’s how to run each module separately.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set global arguments</span></span>
<span><span class="co"># save_dir &lt;- "/Users/hungm/Documents/development/deseq2pip/tests/rnaseq/module/"</span></span>
<span><span class="va">save_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">save_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in dir.create(save_dir, recursive = TRUE): '/tmp/RtmpktrRN0' already</span></span>
<span><span class="co">## exists</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group_by</span> <span class="op">&lt;-</span> <span class="st">"Group2"</span></span>
<span><span class="va">batch</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">org</span> <span class="op">&lt;-</span> <span class="st">"mouse"</span></span>
<span><span class="va">group_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">save_dir</span>, <span class="st">"/"</span>, <span class="va">group_by</span>, <span class="st">"/"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Module 1: Quality Control</span></span>
<span><span class="co">#dds &lt;- run_qc_pip(dds, experiment = "GSE189410", org = org, group_by = group_by, remove_xy = T, remove_mt = T, save_dir = save_dir)</span></span>
<span><span class="co">#dds &lt;- run_dist_pip(dds, experiment = "GSE189410", group_by = group_by, batch = batch, save_dir = save_dir)</span></span>
<span></span>
<span><span class="co"># Module 2: Differential Expression Analysis</span></span>
<span><span class="co">#res &lt;- run_diffexp_pip(dds, org = org, group_by = group_by, save_dir = group_dir)</span></span>
<span></span>
<span><span class="co"># Module 3: GSEA Analysis</span></span>
<span><span class="co">#run_gsea_pip(res = res, org = org, save_dir = group_dir)</span></span></code></pre></div>
<p>Below is a breakdown of the list of processes performed in each
module.</p>
</div>
<div class="section level2">
<h2 id="module-1-quality-control">Module 1: Quality Control<a class="anchor" aria-label="anchor" href="#module-1-quality-control"></a>
</h2>
<p>The <code><a href="../reference/run_qc_pip.html">run_qc_pip()</a></code> module contains a sequence of
subprocesses that performs quality control checks on the data.</p>
<div class="section level3">
<h3 id="remove-xy-and-mt-genes">Remove XY and MT genes<a class="anchor" aria-label="anchor" href="#remove-xy-and-mt-genes"></a>
</h3>
<p>If your data contains samples taken from both males and females that
you are not interested in, you can optionally remove gender-specific
genes on X and Y chromosomes. This can be done with the
<code><a href="../reference/remove_xy_genes.html">remove_xy_genes()</a></code> function.</p>
<p>This is optional but recommended if you see a gender-specific effect
in downstream PCA and distance analysis.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove XY genes (Optional)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_xy_genes.html">remove_xy_genes</a></span><span class="op">(</span><span class="va">dds</span>, org <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removing 4050 XY genes out of 54513 total genes...</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span></span></code></pre></div>
<pre><code><span><span class="co">## class: DESeqDataSet </span></span>
<span><span class="co">## dim: 50463 11 </span></span>
<span><span class="co">## metadata(1): version</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(50463): 0610005C13Rik 0610006L08Rik ... snoZ159 snoZ196</span></span>
<span><span class="co">## rowData names(1): gene</span></span>
<span><span class="co">## colnames(11): BM_IgA_1 BM_IgA_2 ... BM_IgM_3 BM_IgM_4</span></span>
<span><span class="co">## colData names(6): sample Group1 ... sizeFactor condition</span></span></code></pre>
<p>Similarly, if your data contains mitochondrial genes that you are not
interested in, you can optionally remove them with the
<code><a href="../reference/remove_mt_genes.html">remove_mt_genes()</a></code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove MT genes (Optional)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_mt_genes.html">remove_mt_genes</a></span><span class="op">(</span><span class="va">dds</span>, org <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removing 37 MT genes out of 50463 total genes...</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>, <span class="st">"^[Mm][Tt]-"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] FALSE</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span></span></code></pre></div>
<pre><code><span><span class="co">## class: DESeqDataSet </span></span>
<span><span class="co">## dim: 50426 11 </span></span>
<span><span class="co">## metadata(1): version</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(50426): 0610005C13Rik 0610006L08Rik ... snoZ159 snoZ196</span></span>
<span><span class="co">## rowData names(1): gene</span></span>
<span><span class="co">## colnames(11): BM_IgA_1 BM_IgA_2 ... BM_IgM_3 BM_IgM_4</span></span>
<span><span class="co">## colData names(6): sample Group1 ... sizeFactor condition</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="remove-lowly-expressed-genes">Remove lowly expressed genes<a class="anchor" aria-label="anchor" href="#remove-lowly-expressed-genes"></a>
</h3>
<p>RNA-seq is a high-throughput technology that can be noisy, and
certain genes are poorly captured from RNA-seq. To control for this, we
will remove these genes to improve statistical power for subsequent
analysis.</p>
<p>By default we keep genes with at least 5% of total expression
(<code>remove_low_expression(..., quantile = 0.05)</code>). The gene
must also be expressed in at least n replicates, where n is the minimum
replicates in the group.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove low expression genes</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_low_expression.html">remove_low_expression</a></span><span class="op">(</span><span class="va">dds</span>, quantile <span class="op">=</span> <span class="fl">0.05</span>, group_by <span class="op">=</span> <span class="st">"Group2"</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Filtering genes with low expressions...</span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="check-library-size-distribution">Check library size distribution<a class="anchor" aria-label="anchor" href="#check-library-size-distribution"></a>
</h3>
<p>Expression of genes may vary between samples, and this may be due to
sample/batch quality issues.</p>
<p>Here we draw a boxplot representing expression level of every gene
and compare the gene expression distribution from each sample using the
<code><a href="../reference/check_library.html">check_library()</a></code> function.</p>
<p>If the boxplots show little variation in expression level across
samples, this indicates that the samples are of high quality and
suitable for differential expression analysis.</p>
<p>Otherwise, one should consider removing the samples with low quality
or perform batch correction downstream to address quality issues. This
will address technical variation between samples/batches and improve the
statistical power for true biological variations.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># check library size distribution</span></span>
<span><span class="fu"><a href="../reference/check_library.html">check_library</a></span><span class="op">(</span><span class="va">dds</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="run-pca-and-distance-analysis">Run PCA and distance analysis<a class="anchor" aria-label="anchor" href="#run-pca-and-distance-analysis"></a>
</h3>
<p>Next we visualize the relationship between samples/groups using PCA
and distance analysis using the <code><a href="../reference/run_pca.html">run_pca()</a></code> and
<code><a href="../reference/run_distance.html">run_distance()</a></code> functions.</p>
<p>If there are true biological variability between groups, replicas
from the same group should cluster closely in the PCA plot and have
lower distance values in the distance heatmap.</p>
<p>Again, one should consider sample/batch quality if the samples do not
separate as expected.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run PCA for the samples with group information</span></span>
<span><span class="va">vsd</span> <span class="op">&lt;-</span> <span class="fu">vst</span><span class="op">(</span><span class="va">dds</span>, blind <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/run_pca.html">run_pca</a></span><span class="op">(</span><span class="va">vsd</span>, group_by <span class="op">=</span> <span class="st">"Group2"</span>, size <span class="op">=</span> <span class="fl">4</span>, save_data <span class="op">=</span> <span class="cn">FALSE</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## using ntop=500 top features by variance</span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-11-1.png" width="700"><img src="rna_guide_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run distance analysis between samples</span></span>
<span><span class="fu"><a href="../reference/run_distance.html">run_distance</a></span><span class="op">(</span><span class="va">vsd</span>, save_data <span class="op">=</span> <span class="cn">FALSE</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-11-3.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="module-2-differential-expression-analysis">Module 2: Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#module-2-differential-expression-analysis"></a>
</h2>
<p>The <code><a href="../reference/run_diffexp_pip.html">run_diffexp_pip()</a></code> module performs statistical
analysis to identify differentially expressed genes between groups using
<code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq2-package.html" class="external-link">DESeq2::DESeq2()</a></code> function, with p-values were generated
using Wald test.</p>
<div class="section level3">
<h3 id="main-function">Main function<a class="anchor" aria-label="anchor" href="#main-function"></a>
</h3>
<p>The main function in the module is the
<code><a href="../reference/run_diffexp_list.html">run_diffexp_list()</a></code> function, which by default will perform
all pairwise comparisons between groups in the DESeq2 object.</p>
<p>This returns a single dataframe all metrics from
<code><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">DESeq2::results()</a></code>, and the differentially expressed genes
for each comparison is separated by <code>comparison</code> column.
Additionally, chromosome loci and functional annotations of each gene
was added to the dataframe using the
<code><a href="https://hungms.github.io/deseq2pip/reference/run_annotation.html">strpip::run_annotation()</a></code> function (see <a href="https://github.com/hungms/strpip" class="external-link uri">https://github.com/hungms/strpip</a>).</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run differential expression analysis</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_diffexp_list.html">run_diffexp_list</a></span><span class="op">(</span><span class="va">dds</span>, group_by <span class="op">=</span> <span class="st">"Group2"</span>, save_data <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running DESeq2 differential expression...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using pre-existing size factors</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## using 'ashr' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.</span></span>
<span><span class="co">##     https://doi.org/10.1093/biostatistics/kxw041</span></span></code></pre>
<pre><code><span><span class="co">## Combining row metadata to results...</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The OmnipathR package is needed for annotation functions. Please</span></span>
<span><span class="co">## install it with: BiocManager::install('OmnipathR')</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  baseMean log2FoldChange     lfcSE        pvalue         padj</span></span>
<span><span class="co">## Ighg2c...1   1.253765e+06       8.790573 0.4207463 5.535472e-101 8.700101e-97</span></span>
<span><span class="co">## Ighg2b...2   1.815476e+06       7.718483 0.5212007  4.875653e-56 3.831532e-52</span></span>
<span><span class="co">## Ighg3...3    3.243143e+05       7.444125 0.6936469  5.313505e-33 1.670247e-29</span></span>
<span><span class="co">## Ighv1-83...4 1.070368e+03       4.965428 0.3844307  1.614707e-40 6.344586e-37</span></span>
<span><span class="co">## Gm24950...5  8.711461e+01      22.079526 3.6103494  3.810199e-10 9.658855e-08</span></span>
<span><span class="co">## Fads3...6    2.726201e+02       3.900797 0.4287384  2.775847e-21 3.116285e-18</span></span>
<span><span class="co">##                  gene comparison      rank</span></span>
<span><span class="co">## Ighg2c...1     Ighg2c IgG_vs_IgM 844.42666</span></span>
<span><span class="co">## Ighg2b...2     Ighg2b IgG_vs_IgM 396.85838</span></span>
<span><span class="co">## Ighg3...3       Ighg3 IgG_vs_IgM 214.22121</span></span>
<span><span class="co">## Ighv1-83...4 Ighv1-83 IgG_vs_IgM 179.73654</span></span>
<span><span class="co">## Gm24950...5   Gm24950 IgG_vs_IgM 154.88952</span></span>
<span><span class="co">## Fads3...6       Fads3 IgG_vs_IgM  68.28876</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generating-ma-plots">Generating MA plots<a class="anchor" aria-label="anchor" href="#generating-ma-plots"></a>
</h3>
<p>The module also includes the <code><a href="../reference/plot_ma_list.html">plot_ma_list()</a></code> function,
which is a wrapper for the <code><a href="../reference/plot_ma.html">plot_ma()</a></code> command to generate MA
plots for all comparisons.</p>
<p>Here we will use <code><a href="../reference/plot_ma.html">plot_ma()</a></code> to generate a single MA plot
for the <code>IgA_vs_IgG</code> comparison as a demonstration.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># generate MA plot for all comparisons; skipped here</span></span>
<span><span class="co">#plot_ma_list(res, save_plot = FALSE)</span></span>
<span></span>
<span><span class="co"># Generate MA plot for the "IgA_vs_IgG" comparison</span></span>
<span><span class="va">res.IgA_vs_IgG</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">comparison</span> <span class="op">==</span> <span class="st">"IgA_vs_IgG"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_ma.html">plot_ma</a></span><span class="op">(</span><span class="va">res.IgA_vs_IgG</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: ggrepel: 1100 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="generating-volcano-plots">Generating volcano plots<a class="anchor" aria-label="anchor" href="#generating-volcano-plots"></a>
</h3>
<p>The module also includes the <code><a href="../reference/plot_volcano_list.html">plot_volcano_list()</a></code>
function, which is a wrapper for the <code><a href="../reference/plot_volcano.html">plot_volcano()</a></code> command
to generate volcano plots for all comparisons.</p>
<p>Here we will use <code><a href="../reference/plot_volcano.html">plot_volcano()</a></code> to generate a single
volcano plot for the <code>IgA_vs_IgG</code> comparison as a
demonstration.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># generate volcano plot for all comparisons; skipped here</span></span>
<span><span class="co">#plot_volcano_list(res, save_plot = FALSE)</span></span>
<span></span>
<span><span class="co"># Generate volcano plot for the "IgA_vs_IgG" comparison</span></span>
<span><span class="va">res.IgA_vs_IgG</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">comparison</span> <span class="op">==</span> <span class="st">"IgA_vs_IgG"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">res.IgA_vs_IgG</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 33847 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 49953 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 37 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 33847 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_point()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 49953 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: ggrepel: 37 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">## increasing max.overlaps</span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="module-3-gsea-analysis">Module 3: GSEA Analysis<a class="anchor" aria-label="anchor" href="#module-3-gsea-analysis"></a>
</h2>
<p>The final <code><a href="../reference/run_gsea_pip.html">run_gsea_pip()</a></code> module performs gene set
enrichment analysis to identify biological pathways and processes using
a list of MSigDB gene set collections (HALLMARK, GOBP, KEGG, REACTOME,
TFT, BIOCARTA). Gene sets must have at least 10 and no more than 1000
genes to be tested.</p>
<div class="section level3">
<h3 id="main-function-1">Main function<a class="anchor" aria-label="anchor" href="#main-function-1"></a>
</h3>
<p>The main function in the module is the <code><a href="../reference/run_gsea_list.html">run_gsea_list()</a></code>
function, which will perform GSEA analysis using the differential
expression results from the <code><a href="../reference/run_diffexp_list.html">run_diffexp_list()</a></code> function.
This is done for all pairwise comparisons between groups specified in
the <code>comparison</code> column.</p>
<p>This returns a list of gseaResult class objects for each gene set
collection and comparison.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run gsea analysis</span></span>
<span><span class="va">gsea.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_gsea_list.html">run_gsea_list</a></span><span class="op">(</span><span class="va">res</span>, org <span class="op">=</span> <span class="st">"mouse"</span>, save_data <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running gene set enrichment analysis...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (33.26% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.36% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">## using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span></span></code></pre>
<pre><code><span><span class="co">## preparing geneSet collections...</span></span></code></pre>
<pre><code><span><span class="co">## GSEA analysis...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (35.64% of the list).</span></span>
<span><span class="co">## The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre>
<pre><code><span><span class="co">## leading edge analysis...</span></span></code></pre>
<pre><code><span><span class="co">## done...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsea.list</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "IgA_vs_IgG_BIOCARTA" "IgA_vs_IgG_GOBP"     "IgA_vs_IgG_HALLMARK"</span></span>
<span><span class="co">##  [4] "IgA_vs_IgG_KEGG"     "IgA_vs_IgG_REACTOME" "IgA_vs_IgG_TFT"     </span></span>
<span><span class="co">##  [7] "IgA_vs_IgM_BIOCARTA" "IgA_vs_IgM_GOBP"     "IgA_vs_IgM_HALLMARK"</span></span>
<span><span class="co">## [10] "IgA_vs_IgM_KEGG"     "IgA_vs_IgM_REACTOME" "IgA_vs_IgM_TFT"     </span></span>
<span><span class="co">## [13] "IgG_vs_IgM_BIOCARTA" "IgG_vs_IgM_GOBP"     "IgG_vs_IgM_HALLMARK"</span></span>
<span><span class="co">## [16] "IgG_vs_IgM_KEGG"     "IgG_vs_IgM_REACTOME" "IgG_vs_IgM_TFT"</span></span></code></pre>
<p>When <code>save_data = TRUE</code>, individual gsea results that were
saved in the <code>gsea_results</code> directory can be read back in
using the <code><a href="../reference/read_gsea_tsv_list.html">read_gsea_tsv_list()</a></code> function.</p>
<p>Here we will retrieve the gsea results for the
<code>IgA_vs_IgG</code> comparison using GOBP collection.</p>
<div class="sourceCode" id="cb188"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read gsea results from saved files</span></span>
<span><span class="co">#gsea.df &lt;- read_gsea_tsv_list()</span></span>
<span><span class="co">#gsea.df %&gt;% head(.)</span></span>
<span></span>
<span><span class="co"># read gsea results for the "IgA_vs_IgG" comparison</span></span>
<span><span class="va">gsea.IgA_vs_IgG.gobp</span> <span class="op">&lt;-</span> <span class="va">gsea.list</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsea.list</span><span class="op">)</span> <span class="op">==</span> <span class="st">"IgA_vs_IgG_GOBP"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">result</span></span>
<span><span class="va">gsea.IgA_vs_IgG.gobp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                                      ID</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                                             GOBP_HYPERSENSITIVITY</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                                                   GOBP_CELL_DIVISION</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                                 GOBP_CHROMOSOME_SEGREGATION</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                               GOBP_CHROMOSOME_ORGANIZATION</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION     GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION</span></span>
<span><span class="co">##                                                                             Description</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                                             GOBP_HYPERSENSITIVITY</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                                                   GOBP_CELL_DIVISION</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                                 GOBP_CHROMOSOME_SEGREGATION</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                               GOBP_CHROMOSOME_ORGANIZATION</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION     GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION</span></span>
<span><span class="co">##                                             setSize enrichmentScore       NES</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                            16      -0.9695308 -1.884732</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                              607      -0.6694420 -1.950377</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                     405      -0.7144450 -2.019979</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                    602      -0.6489753 -1.887641</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION       129      -0.8214513 -2.126956</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION     118      -0.8197642 -2.104635</span></span>
<span><span class="co">##                                                   pvalue    p.adjust</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                       1.080990e-06 0.005698980</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                          3.481226e-06 0.009176512</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                 7.725985e-06 0.011781740</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                8.939105e-06 0.011781740</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION   2.686117e-05 0.023923555</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION 3.163997e-05 0.023923555</span></span>
<span><span class="co">##                                                  qvalue rank</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                       0.005494843   23</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                          0.008847811 2719</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                 0.011359720 2030</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                0.011359720 2272</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION   0.023066618 2675</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION 0.023066618 2675</span></span>
<span><span class="co">##                                                              leading_edge</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                       tags=31%, list=0%, signal=31%</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                          tags=24%, list=7%, signal=22%</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                 tags=29%, list=5%, signal=27%</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                tags=26%, list=6%, signal=25%</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION   tags=42%, list=7%, signal=39%</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION tags=43%, list=7%, signal=40%</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      core_enrichment</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Il20rb/Ighg3/Ighg2b/Ighg2c/Ighg1</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                                                                                  Calm2/Vrk1/Pdgfc/Ckap5/Zwint/Mastl/Arl8b/Dynlt1b/Ccnk/Cul7/Haus3/Mcmbp/Pdgfd/Bub3/Vegfa/Rhoc/Haus1/Kntc1/Pik3cb/Ncapd2/Nsun2/Ndc80/Fgf2/Kif2a/Tubb5/Sfn/Ska2/Anapc4/Cdc25a/Ruvbl1/Hnrnpu/Kif23/Zfp207/Pkp4/Gnl3/Etv5/Map9/Brca2/Evi5/Kif18b/Bub1b/Smc4/Notch1/Kif14/Brip1/Cetn3/Tnks/Ncapg/Apc/Mapre1/Ube2s/Sgo1/Nusap1/Cenps/Pin1/Rhob/Kif20a/Prc1/Igf1r/Tipin/Orc4/Bora/Usp44/Bcar1/Ncapd3/Exoc6b/Syce2/Cdc25c/Sapcd2/Cdc7/Rcc1/Top1/Tuba1c/Hepacam2/Kif2c/Iqgap3/Itgb3bp/Cep55/Kif20b/Lef1/Nsl1/Zwilch/Tuba1b/Ttc19/Lig1/Fam83d/Bcl2l1/Cdca2/Ran/Top2a/Myc/Map10/Ccnb2/Cdc20/Aspm/Rb1/Pard3b/Incenp/Aurkb/Timeless/Spdl1/E2f8/Nek2/Stmn1/Aurka/Cenpw/Ube2c/Cenpa/Smc2/Plk1/Mis18bp1/Cks1b/Ccna2/E2f7/Spc25/Pmf1/Birc5/Fbxo5/Knstrn/Cdca3/Kif11/Cdca8/Ska1/Nek6/Ccnb1/Sgo2a/Spc24/Bub1/Tacc3/Cenpf/Cdk1/Cdca5/Ncaph/Nuf2/Cks2/Cenpe/Tpx2/Cdc6/Knl1/Tgfb2/Oip5/Ccnd2/Espl1/Ppbp</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                                                                                                                                                                                                                                        Ndc80/Kif2a/Tubb5/M1ap/Ska2/Mlh3/Brca1/Anapc4/Senp6/Hnrnpu/Kif23/Hjurp/Zfp207/Hspa1a/Map9/Fancd2/Kif18b/Bub1b/Kif15/Mre11a/Smc4/Dusp1/Kif14/Brip1/Tnks/Cenpp/Tent4a/Ncapg/Smarca2/Apc/Mapre1/Cenpu/Sgo1/Nusap1/Cenps/Prc1/Arid1b/Rrs1/Psmg2/Uhrf1/Esco2/Usp44/Cenpn/Ncapd3/Syce2/Rcc1/Top1/Trip13/BC055324/Kif2c/Itgb3bp/Cep55/Tubgcp5/Mki67/Nsl1/Cenpm/Zwilch/Xrcc3/Fam83d/Cdca2/Ran/Top2a/Map10/Ccnb2/Cdc20/Aspm/Rb1/Ttk/Incenp/Cep85/Aurkb/Spdl1/Mybl2/Nek2/Aurka/Cenpw/Kif22/Ube2c/Rmi2/Smc2/Plk1/Spc25/Pmf1/Birc5/Ndc1/Fbxo5/Sass6/Knstrn/Dlgap5/Kif11/Cdca8/Ska1/Nek6/Ccnb1/Sgo2a/Spc24/Bub1/Tacc3/Cenpf/Cdk1/Cdca5/Cenpk/Ncaph/Nuf2/Cenpe/Tpx2/Prickle1/Cdc6/Knl1/Kif18a/Cenph/Oip5/Diaph3/Stil/Espl1/Msh5</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                H1f3/Dclre1c/Nav2/Uchl5/Hat1/Kntc1/Ncapd2/H1f4/Ndc80/Kif2a/Naa50/H3c11/Mcm2/Mcm6/Ska2/Cct3/Cct5/Mlh3/Anapc4/Ruvbl1/Senp6/Hnrnpu/Kif23/Cct2/Hjurp/Cct6a/Zfp207/H4c3/Hspa1a/Rpa3/Gnl3/Rfc5/Map9/Rtel1/Fancd2/Brca2/Kif18b/H4c4/Bub1b/Kif15/Mre11a/Smc4/Ssbp1/Dusp1/Trex1/Kif14/Zgrf1/Brip1/Zmpste24/Tnks/Cenpp/Sub1/Tent4a/Ncapg/Smarca2/Apc/Mapre1/Hmgb1/Rfc3/Nsmce1/Cct8/Sgo1/Nusap1/Slx4/Exo1/Prc1/Arid1b/Rrs1/Gins1/Psmg2/Uhrf1/Esco2/Usp44/Cenpn/Ncapd3/Polq/Hnrnpa2b1/Syce2/Apex1/Top1/Mcm4/Trip13/Nhp2/Tcp1/Kif2c/Cip2a/Itgb3bp/Cep55/Nsl1/Hnrnpa1/Pif1/Zwilch/Xrcc3/Hmgb2/Cdc45/Ran/Top2a/Myc/Map10/Mcm8/Cdc20/Rb1/Ttk/Incenp/Dscc1/Aurkb/Spdl1/Rfc4/Mybl2/Nek2/Cenpw/Kif22/Ube2c/Rmi2/Mcm5/Cenpa/Smc2/Plk1/Spc25/Birc5/Ndc1/Fbxo5/Knstrn/Dlgap5/Dna2/Mcm3/Kif11/Cdca8/Ska1/Nek6/Ccnb1/Sgo2a/Rad54l/Spc24/Bub1/Tacc3/H1f5/Cenpf/Cdk1/Cdca5/Cenpk/Ncaph/Rad51/Nuf2/Cenpe/Tpx2/Prickle1/Cdc6/Gins2/Knl1/Kif18a/Cenph/Oip5/Espl1/Msh5</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Zwint/Gen1/Haspin/Bub3/Kntc1/Ncapd2/Ndc80/Anapc4/Hnrnpu/Zfp207/Bub1b/Smc4/Dusp1/Ncapg/Smarca2/Apc/Arid1b/Psmg2/Usp44/Ncapd3/Trip13/Kif2c/Mki67/Zwilch/Xrcc3/Cdca2/Cdc20/Rb1/Ttk/Incenp/Aurkb/Spdl1/Ube2c/Rmi2/Smc2/Plk1/Spc25/Birc5/Fbxo5/Dlgap5/Cdca8/Nek6/Ccnb1/Spc24/Bub1/Tacc3/Cenpf/Cdk1/Ncaph/Nuf2/Cenpe/Cdc6/Knl1/Espl1</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Zwint/Gen1/Cul7/Haspin/Bub3/Kntc1/Rgcc/Ndc80/Cav2/Zfp207/Mtbp/Smpd3/Bub1b/Dusp1/Apc/Nusap1/Pin1/Psmg2/Bora/Usp44/Cdc25c/Rcc1/Trip13/Mki67/Kif20b/Zwilch/Xrcc3/Cdca2/Cdc20/Rb1/Ttk/Incenp/Aurkb/Spdl1/Nek2/Aurka/Ube2c/Plk1/Chek1/Spc25/Birc5/Fbxo5/Dlgap5/Cdca8/Ccnb1/Spc24/Bub1/Cenpf/Nuf2/Knl1/Espl1</span></span>
<span><span class="co">##                                             collection comparison</span></span>
<span><span class="co">## GOBP_HYPERSENSITIVITY                             GOBP IgA_vs_IgG</span></span>
<span><span class="co">## GOBP_CELL_DIVISION                                GOBP IgA_vs_IgG</span></span>
<span><span class="co">## GOBP_CHROMOSOME_SEGREGATION                       GOBP IgA_vs_IgG</span></span>
<span><span class="co">## GOBP_CHROMOSOME_ORGANIZATION                      GOBP IgA_vs_IgG</span></span>
<span><span class="co">## GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION         GOBP IgA_vs_IgG</span></span>
<span><span class="co">## GOBP_REGULATION_OF_MITOTIC_NUCLEAR_DIVISION       GOBP IgA_vs_IgG</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generate-barplots">Generate barplots<a class="anchor" aria-label="anchor" href="#generate-barplots"></a>
</h3>
<p>The module also includes the <code><a href="../reference/plot_gsea_barplot_list.html">plot_gsea_barplot_list()</a></code>
function, which is a wrapper for the <code><a href="../reference/plot_gsea_barplot.html">plot_gsea_barplot()</a></code>
command to generate barplots for all comparisons.</p>
<p>Here we will use <code><a href="../reference/plot_gsea_barplot.html">plot_gsea_barplot()</a></code> to generate a
single barplot for the <code>IgA_vs_IgG</code> comparison using GOBP
collection as a demonstration. By default, the barplot will show the top
10 upregulated and downregulated terms.</p>
<div class="sourceCode" id="cb190"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># generate barplot for all comparisons; skipped here</span></span>
<span><span class="co">#plot_gsea_barplot_list(gsea.df, save_plot = FALSE)</span></span>
<span></span>
<span><span class="co"># Generate barplot for the "IgA_vs_IgG" comparison</span></span>
<span><span class="fu"><a href="../reference/plot_gsea_barplot.html">plot_gsea_barplot</a></span><span class="op">(</span><span class="va">gsea.IgA_vs_IgG.gobp</span>, n <span class="op">=</span> <span class="fl">10</span>, save_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in geom_col(aes(stroke = label), width = 0.75, col = "black"):</span></span>
<span><span class="co">## Ignoring unknown aesthetics: <span style="color: #00BB00;">stroke</span></span></span></code></pre>
<p><img src="rna_guide_files/figure-html/unnamed-chunk-17-1.png" width="700"><img src="rna_guide_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="format-files-for-enrichmentmap">Format files for EnrichmentMap<a class="anchor" aria-label="anchor" href="#format-files-for-enrichmentmap"></a>
</h3>
<p>Finally the module will run <code><a href="../reference/format_enrichmentmap.html">format_enrichmentmap()</a></code> which
creates a enrichmentmap directory containing files formated for
EnrichmentMap visualization.</p>
<p>By default, the function will only use the HALLMARK, GOBP, KEGG and
REACTOME collections. We will skip this step here.</p>
<div class="sourceCode" id="cb192"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#format_enrichmentmap(collection = c("HALLMARK", "GOBP", "KEGG", "REACTOME"))</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Miu Shing Hung.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
